l5kit/l5kit/kinematic/ackerman_perturbation.py

```diff
@@ -4,162 +4,108 @@
-            return history_frames.copy(), future_frames.copy()
+        (
+            lateral_offset_m,
+            longitudinal_offset_m,
+            yaw_offset_rad,
+        ) = self.random_offset_generator()
 
         num_history_frames = len(history_frames)
         num_future_frames = len(future_frames)
-        total_trajectory_length = num_history_frames + num_future_frames
-        if total_trajectory_length < 2:  # TODO is this an error?
-            #  we need at least 2 frames to compute speed and steering rate.
-            return history_frames.copy(), future_frames.copy()
 
-        new_trajectory_to_be_smoothed = _get_history_and_future_frames_as_joint_trajectory(
-            history_frames, future_frames
-        )
+        assert num_history_frames > 0, f"Number of history frames: {num_history_frames}"
+        assert num_future_frames > 0, f"Number of future frames: {num_future_frames}"
 
-        # laterally move the anchor frame
-        lateral_translation_offset = get_lateral_offset_at_idx(
-            new_trajectory_to_be_smoothed, num_history_frames - 1, lateral_offset_distance
-        )
+        trajectory = _get_trajectory(history_frames, future_frames)
 
-        new_trajectory_to_be_smoothed[num_history_frames - 1, :2] += lateral_translation_offset
-
-        # laterally rotate the anchor frame
-        new_trajectory_to_be_smoothed[num_history_frames - 1, 2] += yaw_offset_angle
-
-        #  perform ackerman steering model fitting
-        #  TODO(sms): Replace the call below to a cleaned up implementation
-
-        gx = new_trajectory_to_be_smoothed[:, 0].reshape((-1,))
-        gy = new_trajectory_to_be_smoothed[:, 1].reshape((-1,))
-        gr = new_trajectory_to_be_smoothed[:, 2].reshape((-1,))
-        gv = _compute_speeds_from_positions(new_trajectory_to_be_smoothed[:, :2]).reshape((-1,))
-
-        wx = 5 * np.ones(total_trajectory_length)
-        wy = 5 * np.ones(total_trajectory_length)
-        wr = 5 * np.ones(total_trajectory_length)
-        wv = 5 * np.ones(total_trajectory_length)
-        wgx = np.zeros(total_trajectory_length)
-        wgx[[0, num_history_frames - 1, -1]] = 5
-        wgy = np.zeros(total_trajectory_length)
-        wgy[[0, num_history_frames - 1, -1]] = 5
-        wgr = np.zeros(total_trajectory_length)
-        wgr[[0, num_history_frames - 1, -1]] = 5
-        wgv = np.zeros(total_trajectory_length)
-
-        new_xs, new_ys, new_yaws, new_vs = fit_ackerman_model_approximate(
-            gx, gy, gr, gv, wx, wy, wr, wv, wgx, wgy, wgr, wgv
-        )
+        curr_frame_idx = num_history_frames - 1
+        displacements = _compute_displacements(trajectory[curr_frame_idx:, :2])
 
-        new_trajectory = np.array(list(zip(new_xs, new_ys, new_yaws)))
+        # TODO: ackerman lateral & yaw perturbation does not work when EGO slow moving
+        if np.sum(displacements) < self.min_displacement:
+            lateral_offset_m = 0
```

Comment:
can't we just return here?

Reply:
we still want to keep longitudinal perturbation on slow moving / static EGO, it is beneficial for reducing front collisions